UnitTests/PaymentsCoreTests/APIClient_Tests.swift

```diff
@@ -24,47 +42,72 @@ class APIClient_Tests: XCTestCase {
         }
         """
 
-        let mockRequestResponse = RequestResponseMock(
-            request: AccessTokenRequest(clientID: ""),
-            statusCode: 200,
-            responseString: mockSuccessResponse
-        )
+        mockURLSession.cannedURLResponse = successURLResponse
+        mockURLSession.cannedData = jsonResponse.data(using: String.Encoding.utf8)!
+        
+        let accessTokenRequest = AccessTokenRequest(clientID: "")
+
+        apiClient.fetch(endpoint: accessTokenRequest) { result, _ in
+            switch result {
+            case .success(let response):
+                XCTAssertEqual(response.accessToken, "fake-token")
+                XCTAssertEqual(response.nonce, "fake-nonce")
+                XCTAssertEqual(response.scope, "fake-scope")
+                XCTAssertEqual(response.tokenType, "fake-bearer")
+                XCTAssertEqual(response.expiresIn, 1)
+            case .failure(_):
+                XCTFail("Expect success response")
+            }
 
-        URLProtocolMock.requestResponses.append(mockRequestResponse)
+            expect.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+    }
+        
+    func testFetch_whenServerError_returnsConnectionError() {
+        let expect = expectation(description: "should vend error to client call site")
+        let serverError = NSError(
+            domain: URLError.errorDomain,
+            code: NSURLErrorBadServerResponse,
+            userInfo: nil
+        )
 
-        apiClient.fetch(endpoint: mockRequestResponse) { result, _ in
-            guard case .success(let response) = result else {
-                XCTFail("Expect success response")
+        mockURLSession.cannedError = serverError
+        
+        apiClient.fetch(endpoint: fakeRequest) { result, _ in
+            guard case .failure(.connectionIssue) = result else {
+                XCTFail("")
                 return
             }
-
-            XCTAssertEqual(response.accessToken, "fake-token")
             expect.fulfill()
         }
         waitForExpectations(timeout: 1)
     }
 
-    func testFetch_withAccessTokenFailureMockResponse_vendsUnknownError() {
+    func testFetch_whenNoURLResponse_returnsInvalidURLResponseError() {
         let expect = expectation(description: "Get mock response for access token request")
 
-        let mockFailureResponse: String = """
-        {
-            "error": "unsupported_grant_type",
-            "error_description": "unsupported grant_type"
+        mockURLSession.cannedURLResponse = nil
+
```

Comment:
since we're using a shared `mockURLSession` var on this test class, we'll need to update all of its' state on every test to avoid accidentally reusing properties that aren't set

in this test `mockURLSession` will have the `cannedError` and `cannedData` from whatever test from this file was run immediately before this.

Maybe we recreate a new `apiClient` in each test case? since we're not initializing a new `URLSession` object, this wouldn't be expensive and would let us avoid inadvertently sharing state between tests